core/guard/contract.go

```diff
@@ -13,13 +12,51 @@ import (
 	guardPb "github.com/tron-us/go-btfs-common/protos/guard"
 )
 
+func NewFileStatus(session *storage.FileContracts, contracts []*guardPb.Contract, configuration *config.Config) (*guardPb.FileStoreStatus, error) {
+	guardPid, escrowPid, err := getGuardAndEscrowPid(configuration)
+	if err != nil {
+		return nil, err
+	}
+
+	fileStoreMeta := guardPb.FileStoreMeta{
+		RenterPid:        session.Renter.Pretty(),
+		FileHash:         session.FileHash.String(), //TODO need to check
+		FileSize:         10000,                     //TODO need to revise later
```

Comment:
file size i do think need some calculation. session stores each shardsize and total number of shard

Reply:
no, filesize != sum(shardsize) as reed-solomon will technically triple the file size.